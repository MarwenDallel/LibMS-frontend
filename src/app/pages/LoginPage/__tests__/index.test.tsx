import * as React from 'react';
import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';
import { render } from '@testing-library/react';

import { LoginPage } from '..';

const renderWithRouter = (component: JSX.Element) => {
  const history = createMemoryHistory();
  return {
    ...render(<Router history={history}>{component}</Router>),
  };
};

describe('<LoginPage  />', () => {
  it('should match snapshot', () => {
    const loadingIndicator = renderWithRouter(<LoginPage />);
    expect(loadingIndicator.container.firstChild).toMatchSnapshot();
  });
});
