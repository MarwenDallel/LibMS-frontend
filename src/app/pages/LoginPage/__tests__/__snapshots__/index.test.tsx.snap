// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginPage  /> should match snapshot 1`] = `
.c2 {
  background-color: #007ea4;
}

.c1 {
  background-color: #203549;
}

.c4 {
  background-color: #203549;
}

.c3 {
  color: #203549;
}

.c0 {
  background-color: #f6f8fb;
}

<div
  class="c0 d-flex flex-column min-vh-100"
>
  <div
    class="Header__Div-sc-72bdxd-0"
  >
    <nav
      class="c1 navbar navbar-expand navbar-light"
    >
      <div
        class="container"
      >
        <a
          href="/"
        >
          <img
            alt=""
            height="122"
            src="SMU-LOGO.png"
            width="263"
          />
        </a>
      </div>
    </nav>
    <nav
      class="c2 navbar navbar-expand-md navbar-dark navbar navbar-expand-lg navbar-light"
    >
      <button
        aria-controls="responsive-navbar-nav"
        aria-label="Toggle navigation"
        class="navbar-toggler collapsed"
        type="button"
      >
        <span
          class="navbar-toggler-icon"
        />
      </button>
      <div
        class="navbar-collapse collapse"
        id="responsive-navbar-nav"
      >
        <div
          class="justify-content-center justify-content-md-between container"
        >
          <div
            class="font-weight-light mr-0 text-white"
          >
            My SMU-Library Account
          </div>
        </div>
      </div>
    </nav>
  </div>
  <div
    class="text-left wrapper flex-grow-1 container"
  >
    <div
      class="d-block mt-5 mb-5 font-weight-light"
      style="font-size: 2em;"
    >
      Login
    </div>
    <div
      class="d-block"
      style="font-size: 1.232em;"
    >
      <div
        class="row"
      >
        <div
          class="border-left col"
        >
          <form
            class="mb-3"
            data-testid="login-form"
          >
            <div
              class="form-group"
            >
              <label
                class="form-label"
                for="emailAddr"
              >
                Email address
              </label>
              <input
                class="w-75 form-control"
                id="emailAddr"
                name="email"
                type="email"
              />
              <small
                class="form-text text-muted"
                id="emailHelp"
              >
                We'll never share your email with anyone else.
              </small>
            </div>
            <div
              class="form-group"
            >
              <label
                class="form-label"
                for="password"
              >
                Password
              </label>
              <input
                class="w-75 form-control"
                id="password"
                name="password"
                type="password"
              />
            </div>
            <button
              class="w-75 btn btn-primary"
              type="submit"
            >
              Submit
            </button>
          </form>
        </div>
        <div
          class="border-left col"
        >
          <div
            class="row"
          >
            <div
              class="col"
            >
              <h4>
                Security Notice
              </h4>
              <div
                class="mt-1"
              >
                <small>
                  Logging in will allow you to access SMU Library's resources. Make sure not to share your credentials.
                </small>
                <div>
                  <small>
                    <i
                      class="c3 bi bi-arrow-right-circle-fill mr-2"
                    />
                  </small>
                  <a
                    href="/"
                  >
                    <small>
                      Forgot password?
                    </small>
                  </a>
                </div>
              </div>
            </div>
          </div>
          <div
            class="mt-5 row"
          >
            <div
              class="col"
            >
              <h4>
                Don't have an account?
              </h4>
              <a
                href="/register"
              >
                <button
                  class="btn btn-primary btn-sm"
                  type="button"
                >
                  Register
                </button>
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <footer
    class="c4 text-lg-start"
  >
    <div
      class="container"
    >
      <img
        alt=""
        src="FOOTER-LOGO.png"
      />
    </div>
  </footer>
</div>
`;
